codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/SourceGenerator.scala

```diff
@@ -135,6 +135,9 @@ object SourceGenerator {
       javaOuterClassnameOption = None,
       javaMultipleFiles = false)
 
-  private def nameForMainPackage(model: ModelBuilder.Model): PackageNaming =
-    nameForMainPackage(SourceGeneratorUtils.mainPackageName(model.services.keys ++ model.entities.keys).mkString("."))
+  private def nameForMainPackage(model: ModelBuilder.Model): PackageNaming = {
+    val services = model.services.values.map(_.fqn.fullyQualifiedJavaName)
+    val entities = model.entities.values.map(_.fqn.fullyQualifiedJavaName)
+    nameForMainPackage(SourceGeneratorUtils.mainPackageName(services ++ entities).mkString("."))
```

Comment:
@jrudolph, I'm keeping it as string because we have tests passing string and I didn't want to touch them. 

At the, we only need strings to find out the common package. Passing a FQN won't help much here.

Reply:
Yep, fine for me.